components/header.components.ts

```diff
@@ -0,0 +1,13 @@
+import { Locator, Page } from "@playwright/test";
+
+export class HeaderComponent {
+    readonly signInButton: Locator;
+
+    constructor(private page: Page) {
+        this.signInButton = page.locator(".login")
+    }
+
+    async clickSignInButton(): Promise<void> {
+        await this.signInButton.click();
+    }
+}
```

Comment:
Do we need to use class page object. There is always problem with types.

Reply:
Done